tests/integration/pilot/revisioncmd/revision_view_test.go

```diff
@@ -0,0 +1,360 @@
+	}
+	if !setsMatch(expectedTagSet, actualTagSet) {
+		ctx.Fatalf("tag sets don't match for %s. Expected: %v, Actual:%v", rev, expectedTagSet, actualTagSet)
+	}
+	expectedComponents, ok := expectedComponentsPerRevision[rev]
+	if !ok {
+		ctx.Fatalf("unexpected error. Could not find expected components for %s", rev)
+	}
+	actualComponents := map[string]bool{}
+	for _, iop := range descr.IstioOperatorCRs {
+		for _, comp := range iop.Components {
+			actualComponents[comp] = true
+		}
+	}
+	if !setsMatch(expectedComponents, actualComponents) {
+		ctx.Fatalf("required component sets don't match for revision %s. Expected: %v, Actual: %v",
+			rev, expectedComponents, actualComponents)
+	}
+	verifyComponentPodsForRevision(ctx, "istiod", rev, descr)
+	verifyComponentPodsForRevision(ctx, "ingress-gateway", rev, descr)
+	verifyComponentPodsForRevision(ctx, "egress-gateway", rev, descr)
+}
+
+func verifyComponentPodsForRevision(ctx framework.TestContext, component, rev string, descr *cmd.RevisionDescription) {
+	opComponent := componentMap[component]
+	if opComponent == "" {
+		ctx.Fatalf("unknown component: %s", component)
+	}
+	labelSelector, err := meta_v1.LabelSelectorAsSelector(&meta_v1.LabelSelector{
+		MatchLabels: map[string]string{
+			label.IoIstioRev.Name:        rev,
+			label.OperatorComponent.Name: opComponent,
+		},
+	})
+	if err != nil {
+		ctx.Fatalf("unexpected error: failed to create label selector: %v", err)
+	}
+	componentPods, err := ctx.Clusters().Default().
+		CoreV1().Pods("").
+		List(context.Background(), meta_v1.ListOptions{LabelSelector: labelSelector.String()})
+	if err != nil {
+		ctx.Fatalf("unexpected error while fetching %s pods for revision %s: %v", component, rev, err)
+	}
+	expectedComponentPodSet := map[string]bool{}
+	for _, pod := range componentPods.Items {
+		podName := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+		expectedComponentPodSet[podName] = true
+	}
+
+	actualComponentPodSet := map[string]bool{}
+	podList := []*cmd.PodFilteredInfo{}
+	switch component {
+	case "istiod":
+		podList = descr.ControlPlanePods
+	case "ingress-gateway":
+		podList = descr.IngressGatewayPods
+	case "egress-gateway":
+		podList = descr.EgressGatewayPods
+	}
+	for _, pod := range podList {
+		podName := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+		actualComponentPodSet[podName] = true
+	}
+
+	if !setsMatch(expectedComponentPodSet, actualComponentPodSet) {
+		ctx.Fatalf("%s pods are not listed properly. Expected: %v, Actual: %v",
+			component, expectedComponentPodSet, actualComponentPodSet)
+	}
+}
+
+func setsMatch(expected map[string]bool, actual map[string]bool) bool {
+	if len(expected) != len(actual) {
+		return false
+	}
+	for x := range actual {
+		if !expected[x] {
+			return false
+		}
+	}
+	return true
+}
+
+func subsetOf(a map[string]bool, b map[string]bool) bool {
+	for ax := range a {
+		if !b[ax] {
+			return false
+		}
+	}
+	return true
+}
+
+func skipIfUnsupportedKubernetesVersion(ctx framework.TestContext) {
```

Comment:
Can you please put a comment on why this is needed?

Reply:
Done.

Reply:
Thanks.